haystack/nodes/retriever/multimodal/retriever.py

```diff
@@ -180,17 +177,6 @@ def retrieve_batch(  # type: ignore
                             value range are scaled to a range of [0,1], where 1 means extremely relevant.
                             Otherwise raw similarity scores (for example, cosine or dot_product) are used.
         """
-        filters_list: List[FilterType]
-        if not isinstance(filters, list):
-            filters_list = [filters] * len(queries)
-        else:
-            if len(filters) != len(queries):
-                raise MultiModalRetrieverError(
-                    "The number of filters does not match the number of queries. Provide as many filters "
-                    "as queries, or a single filter that will be applied to all queries."
-                )
```

Comment:
Moved down to `BaseDocumentStore.query_by_embedding_batch` and `SearchEngineDocumentStore.query_by_embedding_batch